### DISCUSSION TITLE
shorebird binaries and integration on build machines

### DISCUSSION THREAD

**Discussion by xalikoutis** [codemagic-ci-cd/codemagic-docs#1787] [Ideas]
### Use case

```Text
build flutter apps with shorebird and create binaries that can be code pushed, bypassing stores update
```

### Idea

```Text
Dont know if you are aware but https://shorebird.dev/ is a company building tools for business that use flutter.
So what shorebird is doing basically is code push for flutter apps. OMG
Please contact them and make possible shorebird to be used easily in codemagic build machines
```

**Comment by Ycoi** [comment]
Thanks for the idea! I have been keeping an eye on them and now when they are about to add official iOS support this indeed could be interesting. Shorebird code push has a CLI but unfortunately currently only supports login via browser. So they would need to add a way to login via the CLI  befofre it can be used in a CI system like Codemagic. 

From curiosity, do you already have experience in using the Shorebird code push and especially I would be interested to know if you have used the iOS beta? And how your experience with it is. 

Do you also happen to know other people who use Shorebird services? Would be nice to know how much interest there would be in using this with Codemagic. That’s something we could then also bring up when discussing with them on a possible integration.

**Comment by xalikoutis** [comment]
Hi for github actions https://docs.shorebird.dev/ci/github

`We've just released Shorebird CLI v0.5.1 🎉

 ✅ release v0 of our [setup-shorebird GitHub action](https://github.com/shorebirdtech/setup-shorebird)

🔑 New shorebird login:ci command to get a SHOREBIRD_TOKEN for CI

📕 New docs for GitHub CI Integration at https://docs.shorebird.dev/ci/github

🍎 Additional iOS preparation

📚 Release notes can be found at https://github.com/shorebirdtech/shorebird/releases/tag/v0.5.1

As always, you can upgrade using shorebird upgrade

Please let us know if we can help!`

**Comment by xalikoutis** [comment]
Also they introduced a new free plan subscription

**Comment by xalikoutis** [comment]
Currently i am working for android push using ` shorebird login:ci` with custom workflow builds.
Only problem is that takes too long to install shorebird.
We have a white label app

**Reply by Ycoi** [reply]
Thanks for the update! How is it going in the mean time? Have you experienced any other issues so far? And how long does the shorebird installation take exactly?

**Reply by Ycoi** [reply]
@xalikoutis Are you still suffering from the issue of long installation times for the shorebird cli? My colleague tested this and for his builds it only takes 36 seconds. 

In the long run we plan to install the shorebird client on our base images, but currently the client gets a lot of updates and very often so we want to wait that this stabilises a bit.

**Reply by xalikoutis** [reply]
Hi sorry for late response.
It is not the shorebird binary itself, but the shorebird flutter version that must be downloaded the first time you use it.
So an easy approach for a build machine is
`curl --proto '=https' --tlsv1.2 https://raw.githubusercontent.com/shorebirdtech/install/main/install.sh -sSf | bash`
`shorebird doctor`

`shorebird doctor` checks for flutter version and downloads it

**Comment by n1kStas** [comment]
I really want this functionality to be in the codemagic service. Manual setup is a bit complicated and takes a lot of time

**Reply by nilsreichardt** [reply]
Hey 👋 There are docs on how to integrate Shorebird + Codemagic. If you have any feedback on how to improve the documentation, please let me know ✌️ 

➡️ https://docs.shorebird.dev/ci/codemagic

**Comment by mikhail-tokarev** [comment]
one more article how to use Shorebird and Codemagic
https://blog.codemagic.io/how-to-set-up-flutter-code-push-with-shorebird-and-codemagic/

**Comment by pushpendraKh** [comment]
Is it possible to use Shorebird with Codemagic workflow not using yaml file?

The only thing we need is to run shorebird release `iOS/android` command instead of flutter build command during build phase.

**Reply by nilsreichardt** [reply]
It is possible, but you need a workaround for that https://github.com/shorebirdtech/docs/issues/76
